eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelType.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.channel
+
+import fr.acinq.eclair.FeatureSupport.Optional
+import fr.acinq.eclair.Features
+import fr.acinq.eclair.Features.{AnchorOutputs, StaticRemoteKey}
+import fr.acinq.eclair.transactions.Transactions.{AnchorOutputsCommitmentFormat, CommitmentFormat, DefaultCommitmentFormat}
+
+/**
+ * Created by t-bast on 24/06/2021.
+ */
+
+/**
+ * A channel type is a specific combination of Bolt 9 features, defined in the RFC (Bolt 2).
+ */
+case class ChannelType(features: Features) {
+
+  val commitmentFormat: CommitmentFormat = {
+    if (features.hasFeature(AnchorOutputs)) {
+      AnchorOutputsCommitmentFormat
+    } else {
+      DefaultCommitmentFormat
+    }
+  }
+
+  /**
+   * True if our main output in the remote commitment is directly sent (without any delay) to one of our wallet addresses.
+   */
+  def paysDirectlyToWallet: Boolean = {
+    features.hasFeature(Features.StaticRemoteKey) && !features.hasFeature(Features.AnchorOutputs)
+  }
+
+}
+
+object ChannelTypes {
```

Comment:
```suggestion
object ChannelType {
```

Reply:
I wanted to be consistent with what we do for features, we have a `Feature` case class and a `Features` object which is a kind of enum that contains `Feature` instances, I think it makes sense?